pinecone/client.go

```diff
@@ -132,6 +389,45 @@ func (c *Client) ListIndexes(ctx context.Context) ([]*Index, error) {
 	return indexes, nil
 }
 
+// CreatePodIndexRequest holds the parameters for creating a new Pods-based index.
+//
+// Fields:
+//   - Name: The name of the index.
+//   - Dimension: The dimension of the index.
+//   - Metric: The metric used to measure the similarity between vectors ("Cosine", "Euclidean", or "Dotproduct").
+//   - Environment: The cloud environment in which the index will be created.
+//   - PodType: The type of pod to use for the index ("p1", "p2", or "s2").
+//   - Shards: The number of shards to use for the index (defaults to 1).
+//   - Replicas: The number of replicas to use for the index (defaults to 1).
+//   - SourceCollection: The Collection from which to create the index.
+//   - MetadataConfig: The metadata configuration for the index.
```

Comment:
Would be nice to include metadata config in the example below.

Reply:
Will do!

Reply:
Done :)